packages/apollo-server-core/src/errors.test.ts

```diff
@@ -170,5 +171,18 @@ describe('Errors', () => {
         name: 'ValidationError',
       });
     });
+    it('provides a BadUserInput error', () => {
+      const error = new BadUserInputError(message, {
+        field1: 'property1',
+        field2: 'property2',
+      });
+      verifyError(error, {
+        code: 'BAD_USER_INPUT',
+        errorClass: BadUserInputError,
+        name: 'BadUserInputError',
+      });
+      expect(error.extensions.exception.field1).to.equal('property1');
```

Comment:
We can make these fields on the root error. The important part is for when these errors are thrown in resolvers to contain the correct layout. [Here](https://github.com/apollographql/apollo-server/pull/1143/files#diff-09c81f433010da1b950ddff2d733546cR71) for example